plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -372,7 +372,9 @@ function perflab_install_and_activate_plugin( string $plugin_slug, array &$proce
 		// Replace new Plugin_Installer_Skin with new Quiet_Upgrader_Skin when output needs to be suppressed.
 		$skin     = new WP_Ajax_Upgrader_Skin( array( 'api' => $plugin_data ) );
 		$upgrader = new Plugin_Upgrader( $skin );
-		$result   = $upgrader->install( $plugin_data['download_link'] );
+		// Remove the version number from the link to download the latest plugin version.
+		$download_link = (string) preg_replace( '#(\/plugin\/[^\/]+)\.\d+\.\d+\.\d+(?:-\w+)?\.zip#', '$1.zip', $plugin_data['download_link'] );
```

Comment:
```suggestion
		$download_link = (string) preg_replace( '#(\/plugin\/[^\/]+)\.\d.*?.zip#', '$1.zip', $plugin_data['download_link'] );
```

Reply:
Or rather, per https://github.com/WordPress/performance/pull/1693#issuecomment-2495000449

```suggestion
		$download_link = (string) preg_replace( '#(\/plugin\/[^\/.]+)\..+?\.zip#', '$1.zip', $plugin_data['download_link'] );
```

Reply:
Stripping off everything up to the first `.` makes sense. However, I also noticed that some plugins have their download link as `https://downloads.wordpress.org/plugin/<plugin-slug>.zip`, particularly those with tags like `trunk`, such as https://downloads.wordpress.org/plugin/better-search-replace.zip.

Given this, I think we might need a regex like this:

```php
$download_link = (string) preg_replace( '#(\/plugin\/[^\.]+).*\.zip#', '$1.zip', $plugin_data['download_link'] );
```

This should account for both cases where the version exists (like `rest-api.2.0-beta15.zip`) and where it doesn't (like `better-search-replace.zip`).

Reply:
On second thought, can’t we just construct the download link directly as:

```php
https://downloads.wordpress.org/plugin/<plugin-slug>.zip
```

This would simplify the logic and remove the need for regex.

Reply:
Hah. Good point.